features/org.wso2.analytics.is.feature/src/main/capp/dashboard/IsAnalytics-Gadget-MessageTable_1.0.0/IsAnalytics_Gadget_MessageTable/js/main.js

```diff
@@ -363,54 +364,94 @@ function getColumns(){
+
+        column["dataKey"] = i;
+        i++;
+        return column;
+    });
 }
 
-function getPdfTableColsAndInfo(){
-    this.getPdfTableColumns=function(columns) {
-        var i = 0;
-        var columns = getColumns();
-        return columns.map(function (column){
-            column["dataKey"] = i;
-            i++;
-            return column;
-        });
+function getPdfTableInfo(maxRecords, totalRecords) {
+
+    var pdfInfo = {};
+    switch (page.name) {
+        case TYPE_OVERALL:
+            pdfInfo["title"] = "OVERALL LOGIN ATTEMPTS";
+            break;
+        case TYPE_LOCAL:
+            pdfInfo["title"] = "LOCAL LOGIN ATTEMPTS";
+            break;
+        case TYPE_FEDERATED:
+            pdfInfo["title"] = "FEDERATED LOGIN ATTEMPTS";
+            break;
+        case TYPE_SESSIONS:
+            pdfInfo["title"] = "LOGIN SESSIONS";
+            break;
+        default:
+            throw "Error - Unknown Page name";
     }
-    this.getPdfTableInfo= function(maxRecords,totalRecords){
-        var pdfInfo = {};
-
-        switch (page.name) {
-                case TYPE_OVERALL:
-                    pdfInfo["title"] = "OVERALL LOGIN ATTEMPTS";
-                    break;
-                case TYPE_LOCAL:
-                    pdfInfo["title"] = "LOCAL LOGIN ATTEMPTS";
-                    break;
-                case TYPE_FEDERATED:
-                    pdfInfo["title"] = "FEDERATED LOGIN ATTEMPTS";
-
-                    break;
-                case TYPE_SESSIONS:
-                    pdfInfo["title"] = "LOGIN SESSIONS REPORT";
-                    break;
-                }
 
-        pdfInfo["headerInfo"] = "Starting Date   : " + renderDateTime(parseInt(listnedTimeFromValue)) + "\n\nEnding Date    : " + renderDateTime(parseInt(listnedTimeToValue)) +
-                                          "\n\nTotal Records : "+totalRecords;
+    pdfInfo["headerInfo"] = "Starting Date   : " + renderDateTime(parseInt(listnedTimeFromValue)) + "\n\nEnding Date    : " + renderDateTime(parseInt(listnedTimeToValue)) + "\n\nTotal Records : " + totalRecords;
+    pdfInfo["fileName"] = pdfInfo.title.toLowerCase().replace(/ /g, "_");
+    pdfInfo["maxRecords"] = maxRecords;
+    pdfInfo["totalRecords"] = totalRecords;
+    return pdfInfo;
+}
 
-        pdfInfo["fileName"]="Logins Report";
-        pdfInfo["maxRecords"]= maxRecords;
-        pdfInfo["totalRecords"]= totalRecords;
-        return pdfInfo;
-    }
+function renderDateTime(data, type, row) {
 
-    function renderDateTime(data, type, row) {
-        var date = new Date(data);
-        return date.toLocaleString("en-US");
-    }
+    var date = new Date(data);
+    return date.toLocaleString(moment.locale());
 }
 
+
+function getPdfTableRows(rawData) {
+
+    return rawData.map(function(record) {
+
+        var columnData = getColumns();
+        var newRecord = {};
+        for (i = 0; i < columnData.length; i++) {
+            if (page.name == TYPE_OVERALL && i == 8) {
```

Comment:
variable i counts the number of columns. 8th column's data comes from the server as "true" or "false". It needs to be converted to "Success" or "Failure" in PDF. That is the way shows in Data Table too. In Data table code also it displays as a static value. 
it is shown in 80-82 lines in [https://github.com/wso2/analytics-is/blob/master/features/org.wso2.analytics.is.feature/src/main/capp/dashboard/IsAnalytics-Gadget-MessageTable_1.0.0/IsAnalytics_Gadget_MessageTable/js/main.js]here. 
I don't see a way to get these values except using static values.